### DISCUSSION TITLE
Adding custom field to logs from filebeat

### DISCUSSION THREAD

**Discussion by Rbartram** [Security-Onion-Solutions/securityonion#5790] [Unsupported Versions] [answered]
Hi.  I have sensor servers running at multiple client locations that are collecting logs using the Fortinet and Azure third party modules.  

I'd like to add customer name and location name fields to all log records based on sensor name.   Should I be looking to modify ingest pipelines or can I add fields by pushing changes to filebeat via minion configs?  I've tried and failed at both, but before troubleshooting further, I'd like guidance on the recommended approach.

Thank You, ...Rob

**Comment by dougburks** [comment, marked as answer]
I'd go with ingest pipelines:
https://docs.securityonion.net/en/2.3/elasticsearch.html#parsing

**Reply by Rbartram** [reply]
Thank you Doug!

**Reply by Rbartram** [reply]
Hi Doug, how does elasticsearch know what pipeline to use?

**Reply by Rbartram** [reply]
I found where the relationships are defined, specifically for azure activitylogs it is defined under module/azure/activitylogs/manifest.yml in the so-filebeat container.  If I wanted to change the ingest-pipeline for development/learning purposes, would I just modify in the so-filebeat container?  Is there way to modify it via salt?

**Reply by dougburks** [reply]
I would NOT recommend modifying containers. You should be able to modify ingest pipelines as mentioned at https://docs.securityonion.net/en/2.3/elasticsearch.html#parsing.

**Reply by Rbartram** [reply]
Thank you Doug!
